holoviews/element/graphs.py

```diff
@@ -422,6 +425,8 @@ def __init__(self, data, kdims=None, vdims=None, **params):
         if isinstance(data, tuple):
             data = data + (None,)*(3-len(data))
             edges, nodes, edgepaths = data
+        elif isinstance(data, type(self)):
+            edges, nodes, edgepaths = data, data.nodes, data._edgepaths
```

Comment:
This duplicates the code you would hope to be handled in ``Graph.__init__`` in the call to ``super``. This suggests some refactoring might be possible but as that might be tricky, it is just a wishlist task and not something to hold up the PR.